app/Metadata/Extractor.php

```diff
@@ -96,15 +96,30 @@ public function extract(string $filename, string $type): array
 			}
 		}
 
+		// Attempt to get sidecar metadata if it exists, make sure to check 'real' path in case of symlinks
+		$sidecarData = [];
+
 		try {
 			// this can throw an exception in the case of Exiftool adapter!
 			$exif = $reader->read($filename);
+
+			$realFile = readlink($filename) ?: $filename;
```

Comment:
What's `?:` doing?

Reply:
it is very similar to `??` but will test on the boolean truth of the value rather than on the isset.
```php
$value ?: $default;
// is equivalent to
!$value ? $value : $default;
// is not equivalent to
(isset($value) && !$value) ? $value : $default;
```
also:
```php
$a = ''; // or 0 or false

$b = $a ?? 'a'; 
// $b is '' or 0 or false

$c = $a ?: 'a';
// $c is 'a'
```

Reply:
Right, basically "use the value from 'readline' if available, otherwise use $filename".

Reply:
Note that readlink "Returns the contents of the symbolic link path or FALSE on error." so `??` would not work here.

Reply:
I'd vote to add a short comment on the line explaining what the notation does to make it easier to read for everyone not being a PHP guru :)